### PR TITLE
Cleanup DLL exports

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

**Comment by jkotas** [comment]
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test please

**Comment by jkotas** [comment]
Addressing https://github.com/dotnet/coreclr/pull/21924#issuecomment-462067477

cc @janvorli @mikedn @am11

**Comment by jkotas** [comment]
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test please

File: src/pal/inc/pal.h [hunk 1]
```diff
@@ -4600,13 +4601,14 @@ PALIMPORT char * __cdecl _strdup(const char *);
...
-PALIMPORT DLLEXPORT void __cdecl qsort(void *, size_t, size_t, int (__cdecl *)(const void *, const void *));
-PALIMPORT void * __cdecl bsearch(const void *, const void *, size_t, size_t,
-int (__cdecl *)(const void *, const void *));
-
 PALIMPORT char * __cdecl _fullpath(char *, const char *, size_t);
 
 #ifndef PAL_STDCPP_COMPAT
+
+PALIMPORT DLLEXPORT void __cdecl qsort(void *, size_t, size_t, int(__cdecl *)(const void *, const void *));
+PALIMPORT DLLEXPORT void * __cdecl bsearch(const void *, const void *, size_t, size_t,
```
Comment (am11) [inline src/pal/inc/pal.h]: Could these be `PALIMPORT PALAPI void bsearch/qsort/getenv/_putenv(..`, as `PALAPI` is defined as `DLLEXPORT __cdecl`?

Comment (jkotas) [reply src/pal/inc/pal.h]: The pattern used to export C-runtime APIs is `PALIMPORT DLLEXPORT`, so I kept this one for consistency. 

C-runtime APIs were not PALAPIs originally when we had a thin (just forwarders) PAL layer for Windows as well. On Windows, the C-runtime APIs were cdecl, but PALAPIs were stdcall.

**Pull request merged**